### ISSUE TITLE
Cannot connect to any saved connections since 1.35.0

### ISSUE THREAD

**Issue by sn4kebite** [microsoft/azuredatastudio#18657]
- Azure Data Studio Version: 1.35.0
- OS Version: Arch Linux (5.16.10-arch1-1)

When trying to connect to any of the saved conections after upgrading to 1.35.0, the server icon briefly shows a spinner then reverts back to the "not connected" state (icon with red dot). Downgrading to 1.34.0 solves the issue. The issue is also present in the insiders build.

When trying to connect from within the query editor (Connect button in the toolbar) it shows the following message:

> Error: The name org.freedesktop.secrets was not provided by any .service files

After enabling this service the connection dialog hangs for a few seconds, then responds with the following

> Error: No such interface 'org.freedesktop.Secret.Collection' at object path '/org/freedesktop/secrets/aliases/default'

FTR I get the following output from keepassxc which provides the org.freedesktop.secrets service while trying to connect in azuredatastudio:

> QDBusConnection: error: could not send reply message to service "": Marshalling failed: Invalid object path passed in arguments
> QDBusConnection: Could not emit signal org.freedesktop.Secret.Service.CollectionChanged: Marshalling failed: Invalid object path passed in arguments

I'm not too familiar with the secrets service, is this now a requirement to use azuredatastudio? I don't see this mentioned in the changelog at all.

Steps to Reproduce:

1. Expand any saved connection in the connections tree.
2. Icon briefly shows a spinner, then reverts.

Does this issue occur when all extensions are disabled?: Yes

**Comment by Charles-Gagnon** [comment]
@abist This looks like it might be caused by the secret storage changes you made.

@sn4kebite The issue seems similar to https://github.com/microsoft/vscode-docker/issues/1515 - in there they mention that installing `gnome-keyring` fixes it. Could you see if that works for you?

**Comment by Charles-Gagnon** [comment]
Thanks for checking. @kburtram FYI for triaging

**Comment by cheenamalhotra** [comment]
Hi @sn4kebite 

Curious to know if the latest release v1.39 is able to unblock you, could you please try the same and confirm?
Thanks for taking time!

**Comment by sn4kebite** [comment]
Without gnome-keyring installed the server icon just shows a spinner. However with gnome-keyring installed I have had no issues for the last few releases that I can recall, so it seems to be working fine.

**Comment by cheenamalhotra** [comment]
Thanks! I will close this then, please feel free to open a new issue if you need additional support :)
